src/controllers/krate/metadata.rs

```diff
@@ -57,17 +56,23 @@ pub fn summary(req: &mut Request) -> CargoResult<Response> {
         .limit(10)
         .load(&*conn)?;
 
-    let recent_downloads = sql::<Nullable<BigInt>>("SUM(crate_downloads.downloads)");
-    let most_recently_downloaded = crates
-        .left_join(
-            crate_downloads::table.on(id.eq(crate_downloads::crate_id)
-                .and(crate_downloads::date.gt(date(now - 90.days())))),
-        )
-        .group_by(id)
-        .order(recent_downloads.desc().nulls_last())
-        .limit(10)
-        .select(ALL_COLUMNS)
-        .load::<Crate>(&*conn)?;
+    // This query needs to be structured in this way to have the LIMIT
+    // happen before the joining/sorting for performance reasons.
+    // It needs to use sql_query because Diesel doesn't have a great way
+    // to join on subselects right now :(
+    let most_recently_downloaded = sql_query(
+        "SELECT crates.* \
+         FROM crates \
+         JOIN ( \
+         SELECT crate_downloads.crate_id, SUM(crate_downloads.downloads) \
+         FROM crate_downloads \
+         WHERE crate_downloads.date > date(CURRENT_TIMESTAMP - INTERVAL '90 days') \
+         GROUP BY crate_downloads.crate_id \
+         ORDER BY SUM(crate_downloads.downloads) DESC NULLS LAST \
+         LIMIT 10 \
+         ) cd ON crates.id = cd.crate_id \
+         ORDER BY cd.sum DESC NULLS LAST",
```

Comment:
I believe NULLS LAST can be dropped because this is an INNER JOIN which should not include any null sums.  Doesn't hurt to be explicit though.